packages/components/src/Checkbox/Checkbox.types.ts

```diff
@@ -0,0 +1,105 @@
+import { ReactElement } from "react";
+import { XOR } from "ts-xor";
+
+export interface BaseCheckboxProps {
+  /**
+   * Determines if the checkbox is checked or not.
+   */
+  readonly checked?: boolean;
+
+  /**
+   * Initial checked value of the checkbox. Only use this when you need to
+   * pre-populate the checked attribute that is not controlled by the component's
+   * state. If a state is controlling it, use the `checked` prop instead.
+   */
+  readonly defaultChecked?: boolean;
+
+  /**
+   * Disables the checkbox.
+   */
+  readonly disabled?: boolean;
+
+  /**
+   * When `true` the checkbox to appears in indeterminate.
+   *
+   * @default false
+   */
+  readonly indeterminate?: boolean;
+
+  /**
+   * Checkbox input name
+   */
+  readonly name?: string;
+
+  /**
+   * Value of the checkbox.
+   */
+  readonly value?: string;
+
+  /**
+   * Further description of the label
+   */
+  readonly description?: string;
+
+  /**
+   * ID for the checkbox input
+   */
+  readonly id?: string;
+
+  /**
+   * Called when the checkbox value changes
+   */
+  onChange?(newValue: boolean): void;
```

Comment:
Do you think it is worth exposing the raw event here. I am thinking it may be useful for the `onFocus` / `onBlur` events

Reply:
yeah I could see the value in that

are you thinking we'd do it for both the existing checkbox and the new one, or just the new one?

Reply:
I think just the new one if that is easier but I could see value in adding to both (if it doesn't blow up scope)

Reply:
gotcha. actually wait, are you saying to add it to just `onChange` or `onChange` `onFocus` and `onBlur`?